docs/source/pages/lightning.rst

```diff
@@ -45,11 +46,42 @@ The example below shows how to use a metric in your `LightningModule <https://py
             self.log('train_acc_epoch', self.accuracy)
 
 .. note::
+
     ``self.log`` in Lightning only supports logging of *scalar-tensors*. While the vast majority of metrics in torchmetrics returns a scalar tensor, some metrics such as
     :class:`~torchmetrics.ConfusionMatrix`, :class:`~torchmetrics.ROC`, :class:`~torchmetrics.MAP`, :class:`~torchmetrics.RougeScore` return outputs that are non-scalar
     tensors (often dicts or list of tensors) and should therefore be dealt with separately. For info about the return type and shape please look at the documentation for
     the ``compute`` method for each metric you want to log.
 
+.. note::
+
+    Modular metrics contain internal states which should belong to only one DataLoader/Dataset. In case you are using multiple dataloaders, it is recommended to initialize a
+    separate modular metric instance for each dataloader and use them separately.
```

Comment:
```suggestion
    Modular metrics contain internal states that belong to only one DataLoader/Dataset. In case you are using multiple dataloaders, it is recommended to initialize a
    separate modular metric instances for each dataloader and use them separately.
```